lib/ddtrace/sampler.rb

```diff
@@ -194,6 +194,12 @@ def key_for(span)
   class PrioritySampler
     extend Forwardable
 
+    # NOTE: We do not advise using a pre-sampler. It can save resources,
+    # but pre-sampling at rates < 100% may result in partial traces, unless
+    # the pre-sampler knows exactly how to drop a span without dropping its ancestors.
+    #
+    # Additionally, as service metrics are calculated in the Datadog Agent,
+    # the service's throughput will be underestimated.
     attr_reader :pre_sampler, :priority_sampler
```

Comment:
I'm actually curious, are there a lot of customers using this feature? Since it ends up being a very sharp edge...

Reply:
I'm not sure it's used in Ruby, but it is used by other language clients.
Because pre-smapler is the only way for the user to reduce traffic between `ddtrace` and the agent, it's used by clients with super high throughput, as their infra wouldn't handle the trace traffic otherwise (mostly due to high load on the trace agent).